File: app/src/main/java/fr/dmconcept/bob/activities/ProjectListActivity.java
package fr.dmconcept.bob.activities;

import android.app.ListActivity;
import android.content.Intent;
import android.os.Bundle;
import android.view.Menu;
import android.view.MenuItem;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.ListView;
import android.widget.TextView;
import fr.dmconcept.bob.BobApplication;
import fr.dmconcept.bob.R;
import fr.dmconcept.bob.models.Project;

import java.util.ArrayList;

public class ProjectListActivity extends ListActivity {

    public static final String TAG = "activities.ProjctListActivity";

    ArrayList<Project> mProjects;

    @Override
    protected void onCreate(Bundle savedInstanceState) {

        super.onCreate(savedInstanceState);

        // Get the project list from the DB
        mProjects = ((BobApplication) getApplication()).getProjectsDao().findAll();

        setListAdapter(new ArrayAdapter<Project>(this, android.R.layout.simple_list_item_2, android.R.id.text1, mProjects) {

            @Override
            public View getView(int position, View convertView, ViewGroup parent) {

                View view = super.getView(position, convertView, parent);

                Project project = getItem(position);

                assert view != null;
                ((TextView) view.findViewById(android.R.id.text1)).setText(project.getName());
                ((TextView) view.findViewById(android.R.id.text2)).setText(project.getBoardConfig().getName());

                return view;
            }

        });

    }

    @Override
    protected void onListItemClick(ListView l, View v, int position, long id) {

        Project p = mProjects.get(position);

        // Start the project details activity
        Intent intent = new Intent(v.getContext(), ProjectActivity.class);
        intent.putExtra(ProjectActivity.EXTRA_PROJECT_ID, p.getId());
        startActivity(intent);
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {

        // Inflate the menu; this adds items to the action bar if it is present.
        getMenuInflater().inflate(R.menu.project_list, menu);

        return super.onCreateOptionsMenu(menu);
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {

        switch (item.getItemId()) {

            case R.id.action_boardConfig:
                menuBoardConfigClicked();
                return true;

            case R.id.action_newProject:
                menuNewProjectClicked();
                return true;

            default:
                return super.onOptionsItemSelected(item);
        }

    }

    /**
     * Show the board config activity
     */
    private void menuBoardConfigClicked() {
        Intent intent = new Intent(this, BoardConfigActivity.class);
        startActivity(intent);
    }

    /**
     * Show the new project activity
     */
    private void menuNewProjectClicked() {
        startActivity(new Intent(this, NewProjectActivity.class));
    }

}

File: bob-client/app/src/main/java/fr/dmconcept/bob/client/activities/activities/ProjectListActivity.scala
package fr.dmconcept.bob.client.activities.activities

import android.app.ListActivity
import android.content.Intent
import android.os.Bundle
import android.util.Log
import android.view.{MenuItem, Menu, ViewGroup, View}
import android.widget.{TextView, ListView, ArrayAdapter}
import fr.dmconcept.bob.client.activities.{NewProjectActivity, BoardConfigActivity, ProjectActivity}
import fr.dmconcept.bob.client.models.Project
import fr.dmconcept.bob.client.{R, BobApplication}
import java.util

object ProjectListActivity {

  val TAG = "activities.ProjctListActivity"

  def log(message: String) = Log.i(TAG, message)

}

class ProjectListActivity extends ListActivity {

  var mApplication: BobApplication = null

  var mProjects: util.List[Project] = null

  override def onCreate(savedInstanceState: Bundle) {

    super.onCreate(savedInstanceState)

    mApplication = getApplication.asInstanceOf[BobApplication]

    // Get the project list from the DB
    mProjects = mApplication.getProjectsDao.findAll()

    // Show the Server IP Selection activity
    startActivity(new Intent(this, classOf[ServerIPSelectionActivity]))

    setListAdapter(new ArrayAdapter[Project](this, android.R.layout.simple_list_item_2, android.R.id.text1, mProjects) {

      override def getView(position: Int, convertView: View, parent: ViewGroup) : View = {

        val view: View = super.getView(position, convertView, parent)

        val project = getItem(position)

        view.findViewById(android.R.id.text1).asInstanceOf[TextView].setText(project.getName)
        view.findViewById(android.R.id.text2).asInstanceOf[TextView].setText(project.getBoardConfig.getName)

        view
      }

    })

  }

  override def onListItemClick(listView: ListView, view: View, position: Int, id: Long) {

    val projectId = mProjects.get(position).getId

    // Start the project details activity
    startActivity(
      new Intent(listView.getContext, classOf[ProjectActivity])
        .putExtra(ProjectActivity.EXTRA_PROJECT_ID, projectId)
    )

  }

  override def onCreateOptionsMenu(menu: Menu): Boolean = {

    // Inflate the menu; this adds items to the action bar if it is present.
    getMenuInflater.inflate(R.menu.project_list, menu)

    super.onCreateOptionsMenu(menu)

  }

  override def onOptionsItemSelected(item: MenuItem): Boolean = {

    item.getItemId match {

      case R.id.action_boardConfig =>
        // Start the "board config" activity
        startActivity(new Intent(this, classOf[BoardConfigActivity]))
        true

      case R.id.action_newProject =>
        // Start the "new project" activity
        startActivity(new Intent(this, classOf[NewProjectActivity]))
        true

      case _ => super.onOptionsItemSelected(item)

    }

  }



}
